core/src/main/java/com/novoda/landingstrip/LandingStrip.java

```diff
@@ -26,6 +29,7 @@
     private final ScrollOffsetCalculator scrollOffsetCalculator;
     private final FastForwarder fastForwarder;
 
+    private Set<ScrollingPageChangeListener> onScrollingPageChangeListeners;
```

Comment:
Yeah they would be added, so it could happen these two things:
- More than one listener is attached when the code says that we only one one and this one is created new every time
- An old listener could receive events when that listener is not valid any more due to new data, if that's the case code might be executed in an invalid state
- We could have a potential memory leak